tap_exchangeratesapi/__init__.py

```diff
@@ -5,18 +5,23 @@
 import argparse
 import time
 import requests
-import singer
 import backoff
 import copy
 
-from datetime import date, datetime, timedelta
+from datetime import datetime, timedelta
+from typing import List, Dict, Any
+
+import singer
+from .discovery import discover
+
 
 base_url = 'https://api.exchangeratesapi.io/'
 
 logger = singer.get_logger()
 session = requests.Session()
 
-DATE_FORMAT='%Y-%m-%d'
+DATE_FORMAT: str = '%Y-%m-%d'
+REQUIRED_CONFIG_KEYS: List[str] = []  # All config keys are optional
```

Comment:
Not polyfill (at least I don't think so). 

Just Python type hints: https://docs.python.org/3/library/typing.html

I'm using python 3.9 but I think this is supported on 3.5 and later. The type hints are prompted by my editor and they help my own sanity while I'm trying to make sure I've got the right types. 

I'll make a note to:
- [ ] remove these type hints before merging, and/or
- [ ] run it through some test suite to ensure python 3.5.2 (and any other required versions) are passing tests.

Reply:
@aaronsteers So, I had a chat about it today, and I had forgotten the Python version is less critical anymore. So if it improves readability, I say keep it as, is, since the goal is teachability.